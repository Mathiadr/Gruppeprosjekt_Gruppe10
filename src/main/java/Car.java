import java.util.ArrayList;

public class Car {
    private String registrationNumber;
    private String owner;
    private String model;
    private String availableDate;   //TODO: IMPLEMENT
    private boolean isAvailable = true; //Whether or not the car is occupied (Not to be confused with available date)
                                        //TODO: IMPLEMENT
    private String fuelType;        //TODO: IMPLEMENT
    private String transmission;    //TODO: IMPLEMENT

    private ArrayList<String> features;


    private static ArrayList<Car> carArrayList = new ArrayList<>();

    public Car(String registrationNumber, String owner, String model) {
        this.registrationNumber = registrationNumber;
        this.owner = owner;
        this.model = model;
    }


    public String getAvailableDate() {
        return availableDate;
    }

    public void setAvailableDate(String availableDate) {
        this.availableDate = availableDate;
    }

    public static ArrayList<Car> getCarArrayList() {
        return carArrayList;
    }

    public static void setCarArrayList(ArrayList<Car> carArrayList) {
        Car.carArrayList = carArrayList;
    }

    public String getRegistrationNumber() {
        return registrationNumber;
    }

    public void setRegistrationNumber(String registrationNumber) {
        this.registrationNumber = registrationNumber;
    }

    public String getOwner() {
        return owner;
    }

    public void setOwner(String owner) {
        this.owner = owner;
    }

    public String getModel() {
        return model;
    }

    public void setModel(String model) {
        this.model = model;
    }
}
