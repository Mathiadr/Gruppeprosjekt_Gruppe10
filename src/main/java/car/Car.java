package car;

import java.util.ArrayList;
import java.util.Collection;

public class Car {
    private String registrationNumber;
    private String owner;
    private String model;
    //private String availableDate;   //TODO: IMPLEMENT
    private boolean isAvailable = true; //Whether the car is occupied (Not to be confused with available date)
                                        //TODO: IMPLEMENT
    //private String fuelType;        //TODO: IMPLEMENT
    //private String transmission;    //TODO: IMPLEMENT

    //private Collection<String> features; //TODO: IMPLEMENT

    public Car(String registrationNumber, String owner, String model) {
        this.registrationNumber = registrationNumber;
        this.owner = owner;
        this.model = model;
    }

    public boolean isAvailable() {
        return isAvailable;
    }

    public void setAvailable(boolean available) {
        isAvailable = available;
    }


    public String getRegistrationNumber() {
        return registrationNumber;
    }

    public void setRegistrationNumber(String registrationNumber) {
        this.registrationNumber = registrationNumber;
    }

    public String getOwner() {
        return owner;
    }

    public void setOwner(String owner) {
        this.owner = owner;
    }

    public String getModel() {
        return model;
    }

    public void setModel(String model) {
        this.model = model;
    }
}
