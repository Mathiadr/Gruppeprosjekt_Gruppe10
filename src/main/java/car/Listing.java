package car;

import java.util.Date;
import java.time.format.DateTimeFormatter;

public class Listing {
        // Contains information of every bid, each of which belongs to one car.
        // TODO: EVERY CAR BELONGS TO ONE LISTING
        // TODO: ur mum
        private String publisherUsername;
        private Date startDate;
        private Date endDate;
        private boolean isAvailable;
        private boolean published;
        private String description;


        public Listing(){}

        public Listing(String publisherUsername, Date startDate, Date endDate, boolean published, String description){
            this.publisherUsername = publisherUsername;
            this.startDate = startDate;
            this.endDate = endDate;
            this.isAvailable = true;
            this.published = published;
            this.description = description;


        }


        public Date getStartDate() {
            return startDate;
        }

        public void setStartDate(Date startDate) {
            this.startDate = startDate;
        }

        public Date getEndDate() {
            return endDate;
        }

        public void setEndDate(Date endDate) {
            this.endDate = endDate;
        }

        public boolean isAvailable() {
            return isAvailable;
        }

        public void setAvailable(boolean available) {
            isAvailable = available;
        }

        public boolean isPublished() {
            return published;
        }

        public void setPublished(boolean published) {
            this.published = published;
        }
}

