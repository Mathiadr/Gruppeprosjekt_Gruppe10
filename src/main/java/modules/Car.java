package modules;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.Date;

public class Car {
    private String registrationNumber;
    private String owner;
    private String model;
    private String brand;                   //TODO: IMPLEMENT
    //private String fuelType;              //TODO: IMPLEMENT
    //private String transmission;          //TODO: IMPLEMENT
    //private Collection<String> features;  //TODO: IMPLEMENT
    private Listing listing;

    public static final String[] fuelTypesList = {"Electric", "Gasoline", "Diesel"};
    public static final String[] transmissionList = {"Manual", "Automatic"};
    public static final String[] featureList = {"GPS", "Air Conditioning"};



    public Car(){

    }

    public Car(String registrationNumber, String owner, String model) {
        this.registrationNumber = registrationNumber;
        this.owner = owner;
        this.model = model;
        LocalDate startDateplaceholder = LocalDate.of(2022, 1, 1);
        LocalDate endDatePlaceholder = LocalDate.of(2022, 12, 31);
        this.listing = new Listing(startDateplaceholder, endDatePlaceholder, true, "");
    }
    public Car(String registrationNumber, String owner, String model, Listing listing) {
        this.registrationNumber = registrationNumber;
        this.owner = owner;
        this.model = model;
        this.listing = listing;
    }


    public String getRegistrationNumber() {
        return registrationNumber;
    }

    public void setRegistrationNumber(String registrationNumber) {
        this.registrationNumber = registrationNumber;
    }

    public String getOwner() {
        return owner;
    }

    public void setOwner(String owner) {
        this.owner = owner;
    }

    public String getModel() {
        return model;
    }

    public void setModel(String model) {
        this.model = model;
    }

    public Listing getListing() {
        return listing;
    }

    public void setListing(Listing listing) {
        this.listing = listing;
    }



    @Override
    public String toString(){
        return "\n\nregistrationNumber: " + registrationNumber + ",\n" +
                "Owner: " + owner + ",\n" +
                "Model: " + model + ",\n" +
                "Listing: " + listing;
    }
}
