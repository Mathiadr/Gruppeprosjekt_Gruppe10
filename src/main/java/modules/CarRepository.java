package modules;

import tools.CarRepositoryFileHandler;

import java.io.File;
import java.util.ArrayList;

public class CarRepository {
    private String repositoryName; // This holds the name of the repository being accessed.
    private ArrayList<Car> carArrayList = new ArrayList<>();
    private final CarRepositoryFileHandler carRepositoryFileHandler = new CarRepositoryFileHandler();

    public CarRepository(String repositoryName) {
        this.repositoryName = repositoryName;

        carArrayList = readFromJSON();
    }



    // Adds a new car to the repository file
    public void AddNewCar(Car car){
        //TODO: implement
        boolean exists = false;
        for (Car i : carArrayList){
            //Checks if the car being added does not already exist within the list
            if (car.getRegistrationNumber().equals(i.getRegistrationNumber())){
                exists = true;
                break;
            }
        }
        if (!exists){
            //System.out.println("Adding car " + car.getRegistrationNumber() + " with car owner " + car.getOwner() + " to carArrayList");
            this.carArrayList.add(car);
        }
        else{
            //System.out.println("Car " + car.getRegistrationNumber() + " Already exists");
        }
    }

    // Removes Car object from carArrayList
    public void RemoveExistingCar(Car car){
        if (carArrayList.contains(car)){
            //System.out.println("Removed car " + car.getRegistrationNumber() + " with car owner " + car.getOwner() + " from carArrayList");
            carArrayList.remove(car);
        }
        else{
            // TODO: Implement error message to GUI
            //System.out.println("Could not find car " + car.getRegistrationNumber() + " in CarArrayList");
        }
    }

    // It rents da car, dumbass
    public void RentCar(Car car){
        // TODO: Test if works
        try{
            for (Car i : GetAllAvailableCars()){
                if (car.getRegistrationNumber().equals(i.getRegistrationNumber())){
                    //TODO: ADD EXTRA MEASURE TO ENSURE IT DOES NOT RENT AN UNAVAILABLE CAR
                    i.getListing().setAvailable(false);
                }
            }
        }
        catch (RuntimeException couldNotRentCar){
            //System.out.println("\n\nCould not rent car " + car.getRegistrationNumber() + "\n\n");
        }

    }

    // Returns all cars that are not currently occupied
    public ArrayList<Car> GetAllAvailableCars(){
        ArrayList<Car> allAvailableCars = new ArrayList<>();
        for (Car i : carArrayList){
            if (i.getListing().isAvailable()){
                allAvailableCars.add(i);
            }
        }
        return allAvailableCars;
    }

    public void updateListing(Car car, Listing listing){
        for (Car i : carArrayList){
            if (car.getRegistrationNumber().equals(i.getRegistrationNumber())){
                i.setListing(listing);
                //System.out.println("Updated listing in Car " + i.getRegistrationNumber());
                break;
            }
        }
    }

    public void SaveCarsToJSON(){
        carRepositoryFileHandler.writeArrayListToFile(getCarArrayList(), repositoryName);
    }

    // Reads all objects from JSON file and maps it to the Car object
    public ArrayList<Car> readFromJSON(){
        return carRepositoryFileHandler.readFromFile(repositoryName);
    }

    public String getRepositoryName() {
        return repositoryName;
    }

    public void setRepositoryName(String repositoryName) {
        this.repositoryName = repositoryName;
    }

    public ArrayList<Car> getCarArrayList() {
        return carArrayList;
    }

    public void setCarArrayList(ArrayList<Car> carArrayList) {
        this.carArrayList = carArrayList;
    }
}
