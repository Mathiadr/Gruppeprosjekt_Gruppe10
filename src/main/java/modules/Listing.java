package modules;


import java.util.Date;

public class Listing {
    // Contains information of every bid, each of which belongs to one car.
    // TODO: EVERY CAR BELONGS TO ONE LISTING
    // TODO: ur mum
    private String publisherUsername;
    private Date startDate;
    private Date endDate;
    private Car car;
    private boolean isAvailable;
    private boolean published;
    private String description;

    public Listing(){}

    public Listing(String publisherUsername, Date startDate, Date endDate, Car car, boolean published, String description){
        this.publisherUsername = publisherUsername;
        this.startDate = startDate;
        this.endDate = endDate;
        this.car    = car;
        this.isAvailable = true;
        this.published = published;
        this.description = description;
    }


    public Date getStartDate() {
        return startDate;
    }

    public void setStartDate(Date startDate) {
        this.startDate = startDate;
    }

    public Date getEndDate() {
        return endDate;
    }

    public void setEndDate(Date endDate) {
        this.endDate = endDate;
    }

    public Car getCar() {
        return car;
    }

    public void setCar(Car car) {
        this.car = car;
    }

    public boolean isAvailable() {
        return isAvailable;
    }

    public void setAvailable(boolean available) {
        isAvailable = available;
    }

    public boolean isPublished() {
        return published;
    }

    public void setPublished(boolean published) {
        this.published = published;
    }



}
