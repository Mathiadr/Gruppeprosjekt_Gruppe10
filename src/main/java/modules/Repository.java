package modules;

import tools.RepositoryFileHandler;

import java.util.ArrayList;
import java.util.Objects;

public class Repository {
    private String repositoryName; // This holds the name of the repository being accessed.
    private ArrayList<Listing> listingArrayList = new ArrayList<Listing>();

    private final RepositoryFileHandler repositoryFileHandler = new RepositoryFileHandler();

    public Repository(String repositoryName) {
        this.repositoryName = repositoryName;

        listingArrayList = readFromJSON();
    }



    // Reads all objects from JSON file and returns an ArrayList of Listing objects
    public ArrayList<Listing> readFromJSON(){
        return repositoryFileHandler.readFromFile(repositoryName);
    }


    public void addNewListing(Listing listing){
        boolean exists = false;
        for (Listing i : listingArrayList){
            //TODO: IMPLEMENT
            if(Objects.equals(i.getCar().getRegistrationNumber(), listing.getCar().getRegistrationNumber())){
                exists = true;
                break;
            }
        }
        if(!exists){
            listingArrayList.add(listing);
        }

    }

    // Removes Car object from listingArrayList
    public void removeExistingListing(Car car){
        // TODO: IMPLEMENT
    }

    // It rents da car, dumbass
    public void acceptListing(Car car){
        //TODO: REMOVE & RENAME
    }

    // Returns all cars that are not currently occupied
    public ArrayList<Listing> GetAllAvailableListings(){
        // TODO: Add date as parameter argument
        ArrayList<Listing> allAvailableListings = new ArrayList<>();
        for (Listing i : listingArrayList){
            if (i.isAvailable()){
                allAvailableListings.add(i);
            }
        }
        return allAvailableListings;
    }

    public void SaveCarsToJSON(){
        repositoryFileHandler.writeArrayListToFile(getListingArrayList(), repositoryName);
    }

    public String getRepositoryName() {
        return repositoryName;
    }

    public void setRepositoryName(String repositoryName) {
        this.repositoryName = repositoryName;
    }

    public ArrayList<Listing> getListingArrayList() {
        return listingArrayList;
    }

    public void setListingArrayList(ArrayList<Listing> listingArrayList) {
        this.listingArrayList = listingArrayList;
    }
}
