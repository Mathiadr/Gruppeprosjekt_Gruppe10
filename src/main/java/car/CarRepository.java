package car;

import org.json.JSONObject;

import java.io.*;
import java.util.ArrayList;

public class CarRepository {
    private String repositoryName;

    private static ArrayList<Car> availableCarArrayList; // Holds cars not currently occupied
    private static ArrayList<Car> allCarArrayList; // Holds ALL cars. Unsure if necessary

    CarRepository(String repositoryName) throws FileNotFoundException {
        this.repositoryName = repositoryName;

        FileReader fileReader = new FileReader(this.repositoryName);
    }

    // Adds a new car to the repository file
    public void AddNewCar(Car car){
        //TODO: implement
        JSONObject jsonObject = new JSONObject(car.getRegistrationNumber());
        jsonObject.put("registrationNumber", car.getRegistrationNumber());
        jsonObject.put("owner", car.getOwner());
        jsonObject.put("model", car.getModel());
        jsonObject.put("availableDate", car.getAvailableDate());
        jsonObject.put("isAvailable", car.isAvailable());
        jsonObject.put("fuelType", car.getFuelType());
        jsonObject.put("transmission", car.getTransmission());
        jsonObject.put("features", car.getFeatures()); //Maybe a nested JSONObject/JSONArray belongs here?
        try{
            FileWriter fileWriter = new FileWriter(this.repositoryName);
            fileWriter.append(jsonObject.toString());
            System.out.println("Car " + car.getRegistrationNumber() + " has been written to file.");
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    // It rents da car, dumbass
    public void RentCar(Car car){
        //TODO: implement
    }

    // Returns all cars that are not currently occupied
    public ArrayList<Car> GetAllAvailableCars(){
        //TODO: implement
        return null;
    }

    // Unsure what to do here...
    public void SaveAllAvailableCars(){
        //TODO: implement
    }


}
