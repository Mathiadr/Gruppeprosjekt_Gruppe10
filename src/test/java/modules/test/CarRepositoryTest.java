package modules.test;

import modules.Car;
import modules.CarRepository;
import modules.Listing;
import org.approvaltests.Approvals;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.Objects;

import static org.junit.jupiter.api.Assertions.*;

public class CarRepositoryTest {

    CarRepository carRepository;

    @BeforeAll
    static void initializeCarRepository(){
        CarRepository carRepository = new CarRepository("testRepository.json");
        carRepository.addNewCar(
                new Car("MXH10045", "Mathias Dale Ratdal", "E2",
                        new Listing(
                                LocalDate.of(2022, 1, 1),
                                LocalDate.of(2022, 3, 7),
                                true, "PLACEHOLDER DESCRIPTION")));
        carRepository.addNewCar(
                new Car("EBX23365", "Sokrat Starynkiewicz", "A4",
                        new Listing(
                                LocalDate.of(2022, 2, 25),
                                LocalDate.of(2022, 6, 7),
                                true, "PLACEHOLDER DESCRIPTION")));
        carRepository.addNewCar(
                new Car("E0B10045", "Joe Mama", "C2",
                        new Listing(
                                LocalDate.of(2022, 2, 1),
                                LocalDate.of(2022, 2, 10),
                                true, "PLACEHOLDER DESCRIPTION")));
        carRepository.addNewCar(
                new Car("ADHD0O11", "Sander Kristiansen", "C6",
                        new Listing(
                                LocalDate.of(2022, 7, 15),
                                LocalDate.of(2022, 8, 15),
                                true, "PLACEHOLDER DESCRIPTION")));
        carRepository.addNewCar(
                new Car("D1CK9864", "Djikstra Van Houven", "BA1",
                        new Listing(
                                LocalDate.of(2022, 10, 14),
                                LocalDate.of(2022, 11, 4),
                                true, "PLACEHOLDER DESCRIPTION")));
        carRepository.addNewCar(
                new Car("P0043456", "John Doe", "A2",
                        new Listing(
                                LocalDate.of(2022, 12, 1),
                                LocalDate.of(2022, 12, 31),
                                true, "PLACEHOLDER DESCRIPTION")));
        carRepository.addNewCar(
                new Car("BE240051", "Ola Nordmann", "A1",
                        new Listing(
                                LocalDate.of(2022, 2, 7),
                                LocalDate.of(2022, 12, 23),
                                true, "PLACEHOLDER DESCRIPTION")));

        carRepository.saveCarsToJSON();
        carRepository.readFromJSON();
    }

    @BeforeEach
    void resetCarRepository(){
        carRepository = new CarRepository("testRepository.json");
        carRepository.readFromJSON();
    }

    @Test
    public void new_car_is_creatable(){
        Car car = new Car("DFG441563", "Dummy one", "Volvo");
        assertNotNull(car);
    }


    @Test
    public void car_is_added_to_repository(){
        Car car = new Car("DFG441563", "Dummy one", "Volvo");
        carRepository.addNewCar(car);
        assertTrue(carRepository.getCarArrayList().contains(car));
    }

    @Test
    public void car_duplicate_is_NOT_added_to_repository(){
        Car car1 = new Car("DFG441563", "Dummy one", "Volvo");
        carRepository.addNewCar(car1);
        carRepository.addNewCar(car1);
        int count = 0;
        for (Car car : carRepository.getCarArrayList()) {
            if (car == car1) count++;
        }
        assertEquals(1, count);
    }

    @Test
    public void car_with_existing_registration_number_is_NOT_added_to_repository(){
        Car car1 = new Car("DFG441563", "Dummy one", "Volvo");
        Car car2 = new Car("DFG441563", "Dummy two", "Toyota");
        carRepository.addNewCar(car1);
        carRepository.addNewCar(car2);
        assertFalse(carRepository.getCarArrayList().contains(car2));
    }

    @Test
    public void car_is_removed_from_repository(){ // TODO: save, Re-read and check
        Car car1 = new Car("DFG441563", "Dummy one", "Volvo");
        ArrayList<Car> carArrayList = carRepository.getCarArrayList();
        carRepository.addNewCar(car1);
        carRepository.removeExistingCar(car1);
        assertFalse(carArrayList.contains(car1), "Specified car was not removed");
    }

    @Test
    public void new_listing_replaces_old_listing(){
        Car car1 = new Car("DFG441563", "Dummy one", "Volvo",
                new Listing(null, null, true, "This is the original listing"));
        carRepository.addNewCar(car1);
        Listing updatedListing = new Listing(null, null, true, "This is the updated listing");
        carRepository.updateListing(car1, updatedListing);
        assertNotEquals("This is the original listing", car1.getListing().getDescription());
        assertEquals("This is the updated listing", car1.getListing().getDescription());
    }


    @Test
    /*
    *
     */
    public void list_of_available_cars_contains_all_available_cars(){
        carRepository.readFromJSON();
        Approvals.verify(carRepository.getAllAvailableCars());
    }

    @Test
    public void list_of_available_cars_does_not_contain_unavailable_cars(){
        carRepository.getCarArrayList().get(0).getListing().setAvailable(false);
        ArrayList<Car> availableCarList = carRepository.getAllAvailableCars();
        assertFalse(availableCarList.contains(carRepository.getCarArrayList().get(0)));
        Approvals.verify(carRepository.getAllAvailableCars());

    }

    @Test
    public void can_rent_car_if_within_rentable_period(){
        LocalDate listingStartDate = LocalDate.of(2022, 1, 1);
        LocalDate listingEndDate = LocalDate.of(2022, 6, 6);
        Car car1 = new Car("DFG441563", "Dummy one", "Volvo",
                new Listing(listingStartDate, listingEndDate, true, "TEST DESCRIPTION"));
        carRepository.addNewCar(car1);
        LocalDate startOfRentPeriod = LocalDate.of(2022, 3, 3);
        LocalDate endOfRentPeriod = LocalDate.of(2022, 4, 4);

        assertEquals(1 , carRepository.initiateRentProcess(car1, startOfRentPeriod, endOfRentPeriod));
    }

    @Test
    public void cannot_rent_car_if_outside_of_rentable_period(){
        LocalDate listingStartDate = LocalDate.of(2022, 1, 1);
        LocalDate listingEndDate = LocalDate.of(2022, 6, 6);
        Car car1 = new Car("DFG441563", "Dummy one", "Volvo",
                new Listing(listingStartDate, listingEndDate, true, "TEST DESCRIPTION"));
        carRepository.addNewCar(car1);
        LocalDate startOfRentPeriod = LocalDate.of(2022, 4, 4);
        LocalDate endOfRentPeriod = LocalDate.of(2022, 7, 7);

        assertEquals(-1 , carRepository.initiateRentProcess(car1, startOfRentPeriod, endOfRentPeriod));
    }

    @Test
    public void cannot_rent_car_if_illogical_date(){
        LocalDate listingStartDate = LocalDate.of(2022, 1, 1);
        LocalDate listingEndDate = LocalDate.of(2022, 6, 6);
        Car car1 = new Car("DFG441563", "Dummy one", "Volvo",
                new Listing(listingStartDate, listingEndDate, true, "TEST DESCRIPTION"));
        carRepository.addNewCar(car1);
        LocalDate startOfRentPeriod = LocalDate.of(2022, 5, 5);
        LocalDate endOfRentPeriod = LocalDate.of(2022, 4, 4);

        assertEquals(-1 , carRepository.initiateRentProcess(car1, startOfRentPeriod, endOfRentPeriod));
    }

    @Test
    public void car_is_not_available_after_renting(){
        LocalDate listingStartDate = LocalDate.of(2022, 1, 1);
        LocalDate listingEndDate = LocalDate.of(2022, 6, 6);
        Car newCar = new Car("DFG441563", "Dummy one", "Volvo",
                new Listing(listingStartDate, listingEndDate, true, "TEST DESCRIPTION"));
        carRepository.addNewCar(newCar);
        LocalDate startOfRentPeriod = LocalDate.of(2022, 3, 3);
        LocalDate endOfRentPeriod = LocalDate.of(2022, 4, 4);

        carRepository.initiateRentProcess(newCar, startOfRentPeriod, endOfRentPeriod);

        ArrayList<Car> availableCarList = carRepository.getAllAvailableCars();
        assertFalse(availableCarList.contains(newCar));
    }
}
