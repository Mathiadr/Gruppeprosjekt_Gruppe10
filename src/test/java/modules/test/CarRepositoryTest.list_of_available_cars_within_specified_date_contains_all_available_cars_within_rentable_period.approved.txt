[

registrationNumber: MXH10045,
Owner: Mathias Dale Ratdal,
Model: E2,
Listing:    Start date: 2022-01-01,
   End date: 2022-03-07,
   isAvailable: true,
   Description : PLACEHOLDER DESCRIPTION, 

registrationNumber: EBX23365,
Owner: Sokrat Starynkiewicz,
Model: A4,
Listing:    Start date: 2022-02-25,
   End date: 2022-06-07,
   isAvailable: true,
   Description : PLACEHOLDER DESCRIPTION, 

registrationNumber: BE240051,
Owner: Ola Nordmann,
Model: A1,
Listing:    Start date: 2022-02-07,
   End date: 2022-12-23,
   isAvailable: true,
   Description : PLACEHOLDER DESCRIPTION]